### DISCUSSION TITLE
Filament Runout Sensor

### DISCUSSION THREAD

**Discussion by Lambrecht26** [Jyers/Marlin#846] [Q&A] [answered]
Hi

Got a Ender 3 V2, Creality filament sensor - Jyers 1.3.1.  Works like a charm

Every time i activate the filament sensor, it prints for 2 sec. and then apparantly thinks the filament is out, because it retracts the filament, as it would if theres was i real runout

The sensor has the blue light on when its plugged in

Anyone?

**Comment by Cal-LF** [comment, marked as answer]
The runout length set to 10?
Or the polarity of the switch is reversed.

**Reply by Lambrecht26** [reply]
No, set to 350 I think - thought it should be the lenght of the Bowden tube

**Reply by Cal-LF** [reply]
Filament runout distance is around 10 for me. 
You are referring to the filament load and unload length for change filament. Mine is 455.
If the sensor still triggers, then the switch requires rewiring or compile your own firmware changing the filament sensor state from low to high.
Refer to Issues: https://github.com/Jyers/Marlin/issues/843

**Reply by Lambrecht26** [reply]
Tried to swap the wiring, then the printer doesnt boot - i'll try to set it to 10

**Reply by Cal-LF** [reply]
Refer #567 #481

**Reply by Cal-LF** [reply]
Firmware from https://github.com/mriscoc/Marlin_Ender3v2/releases
offers menu setup of sensor plus features similar to Jyres.

**Reply by Jyers** [reply]
Be careful how you switch the wiring, as you might need to do it on the switch side for it to work. What type of sensor are you using?

**Reply by Lambrecht26** [reply]
This:. https://3deksperten.dk/creality-3d-filament-sensor-switch-with-bracket.html

The dealer who sold me the printer had Filamentsensor in the description og the printer, so I held Them Up to that, and they send me this

**Reply by adam3654** [reply]
I just installed a filament sensor using the z end stop switch Saturday night, I traced the pins and realized that if I wired everything as labeled (V-->V, G-->G, S-->PA-4) when the filament was present, the firmware would think it was in a runout state. I had to wire from the ground pin on the motherboard to the V pin on the sensor. I didn't connect the V pin on either end because PA-4 is already pulled up. The switch I used has a place for an led but it's unused. If the same switch is used on your sensor, you could probably hack it to make it work, losing the led functionality, or recompile the firmware for a low state normal.

**Comment by smooreace** [comment]
Im having strange run-out sensor issues too... trigger the sensor and the
printer soft-reboots.

**Reply by vali20** [reply]
I had the same problem, i had outer pins of the sensor inversed. Take a look at #578 

Also, I think the distance you have to set is distance between the sensor and the extruder, not the length of bowden tube.

**Comment by smooreace** [comment]
Thanks, Ill take a look. I could be using the wrong 2 pins, but they cant
be reversed.. its a simple switch. Off or on.

**Reply by vali20** [reply]
The sensor can have two pins, but on the motherboard there are 3 pins, with the ground in the middle. Did you check the wiring?
Can be very dangerous for motherboard, because in my case, when sensor triggered, the vcc was shorted to ground, instead of signal pin being shorted to ground.
This was the reason for reboots.

**Comment by smooreace** [comment]
Im using the right 2 pins with the board oriented with the extruder ports
at the top

**Reply by vali20** [reply]
Let me show you more clearly. Take a look at the picture and check if you have the same.
When sensor triggers, it should short the red pin out to the black one.
![Filement1](https://user-images.githubusercontent.com/73364623/118590152-9100bc80-b7aa-11eb-9e3b-63b1c4393c20.png)

If you have this sensor, like I do, you should inverse the outer wires and disconnect the white one. The white one is VCC 3,3V, not signal. This is why the printers rebootes, because sensor shorted out the VCC to GND (white one to black one)
![Filament2](https://user-images.githubusercontent.com/73364623/118590264-c1e0f180-b7aa-11eb-80c4-50153e42f1c3.png)
P.S. The wiring in the last image (the sensor) is **WRONG** as it is shown here

**Comment by Lambrecht26** [comment]
Didnt help, runout length set to 10, Get a filament runout error, tried to swap the red and yellow(white) wire, that freezes the printer

**Reply by Jyers** [reply]
Did you try what was recommended above? "you should inverse the outer wires and disconnect the white one"

**Reply by Lambrecht26** [reply]
Disconnected my Yellow wire (White in the picture) - looks fine for now

**Reply by Lambrecht26** [reply]
Thanks
